platform.md

```diff
@@ -276,10 +275,79 @@ All lifecycle phases:
 - MUST read `CNB_PLATFORM_API` from the execution environment and evaluate compatibility before attempting to parse other inputs (see [Platform API Compatibility](#platform-api-compatibility))
 - MUST give command line inputs precedence over other inputs
 
+#### `analyzer`
+The platform MUST execute `analyzer` in the **build environment**
+
+Usage:
+```
+/cnb/lifecycle/analyzer \
+  [-analyzed <analyzed>] \
+  [-daemon] \ # sets <daemon>
+  [-gid <gid>] \
+  [-layers <layers>] \
+  [-log-level <log-level>] \
+  [-order <order>] \
+  [-previous-image <previous-image> ] \
+  [-run-image <run-image> ] \
```

Comment:
Do we need `-run-image` if we have `stack`?

Reply:
Also why would we allow run-image to be overridden but not build-image?

Reply:
I think so, yes. Since a run image could exist on any registry, including private registries, I think we want to continue to support taking in a `run-image`. 

Phases after this should maybe be able to remove it though, right @ekcasey ?

I think we could remove `<run-image>` from `/builder` entirely since the `/analyzer` will produce an `analyzed.toml` with the image ref for the chosen `<run-image>`.

Reply:
I guess I'm still confused why you couldn't pass in the build-image. What if `/cnb/stack.toml` doesn't contain it? Should the build fail?

Reply:
You can't pass in the build-image because you are running _on_ the build image for analyzer now.

Reply:
@jabrown85 I don't think we should require that you run analyze on the build image, only that `stack.toml` accurately describes the build image.